src/components/Provider.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { createContext, useCallback, useMemo, useRef } from "react";
+
+export const Context = createContext({
+    generateId: () => {
+        throw new Error("Please wrap your app with <Provider /> component from @xola/ui-kit");
+    },
+});
+
+/**
+ * UI Kit's default provider.
+ * Must be used from now on in order to generate correct component IDs.
+ * Also a good place to implement any global state required for the UI Kit in the future.
+ */
+export const Provider = ({ children }) => {
```

Comment:
Can you explain why we need to use context to solve this problem instead of just creating a general hook?

Reply:
Because of React's strict mode (in development). Since strict mode renders the whole app twice, we need to restart the counter to get exact same values in server and client. This provider creates a context that is hooked to a lifecycle of the app, meaning it gets recreated everytime the app is rendered for the first time.

This is how the hook-only version would work:

```tsx
let idCounter = 1;

const useId = () => {
    const [id] = useState(() => idCounter++);
    return id;
};
```

This would be the flow of execution without strict mode:
1. The app renders on the server
1. The component gets the `id={1}`
1. The rendered HTML is sent to the client
1. When JS is loaded, the client re-renders the app
1. The component also gets the `id={1}`

And this is with strict mode:
1. The app renders on the server
1. The component gets the `id={1}`
1. The rendered HTML is sent to the client
1. When JS is loaded, the client re-renders the app in the background and strict mode is checking if everything is ok
1. The app is rendered again to be displayed on the page, so the component gets `id={2}` and now we a mismatch - server didn't render the same this as the client

In the PR the solution is to hold the current ID counter inside the provider component and by using the context to allow the hook to get the next ID in a predictable way.

Btw, this problem would not exists in the production build.